pkg/cmd/prune-backups/prune.go

```diff
@@ -0,0 +1,230 @@
+package prune_backups
+
+import (
+	goflag "flag"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io/fs"
+	"k8s.io/klog/v2"
+	"os"
+	"path"
+	"sort"
+	"time"
+)
+
+// BasePath for Backups we assume we have full ownership over the root folder, for tests we will change this to a tmp directory
+var BasePath = "/"
+
+const RetentionTypeNone = "None"
+const RetentionTypeSize = "RetentionSize"
+const RetentionTypeNumber = "RetentionNumber"
+
+type backupDirStats []backupDirStat
+
+type backupDirStat struct {
+	name      string
+	sizeBytes int64
+	modTime   time.Time
+}
+
+type pruneOpts struct {
+	retentionType      string
+	maxNumberOfBackups int
+	maxSizeOfBackupsGb int
+}
+
+func NewPruneCommand() *cobra.Command {
+	opts := pruneOpts{retentionType: "None"}
+	cmd := &cobra.Command{
+		Use:   "prune-backups",
+		Short: "Prunes existing backups on the filesystem.",
+		Run: func(cmd *cobra.Command, args []string) {
+			defer klog.Flush()
+
+			if err := opts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := opts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	opts.AddFlags(cmd)
+	return cmd
+}
+
+func (r *pruneOpts) AddFlags(cmd *cobra.Command) {
+	flagSet := cmd.Flags()
+	flagSet.StringVar(&r.retentionType, "type", r.retentionType, "Which kind of retention to execute, can either be None, RetentionNumber or RetentionSize.")
+	flagSet.IntVar(&r.maxNumberOfBackups, "maxNumberOfBackups", 1, "how many backups to keep when type=RetentionNumber")
```

Comment:
Not sure if we need to set the default here or in the periodic controller template or both:
https://github.com/openshift/cluster-etcd-operator/blob/381ffb81706699cdadd0735a52f9d20379505ef7/pkg/operator/periodicbackupcontroller/periodicbackupcontroller.go#L145

But per the API the default should be `maxNumberOfBackups: 15`
https://github.com/openshift/api/blob/master/config/v1alpha1/types_backup.go#L115

Reply:
I think we add this to the periodic controller to keep it close to the schema. Thanks for reminding me of the default value again, I was just browsing through the min/max properties